packages/pharos/src/components/sheet/pharos-sheet.ts

```diff
@@ -0,0 +1,367 @@
+ * @fires pharos-sheet-open - Fires when the sheet is about to open - cancelable
+ * @fires pharos-sheet-opened - Fires when the sheet has opened
+ * @fires pharos-sheet-close - Fires when the sheet is about to close - cancelable
+ * @fires pharos-sheet-closed - Fires when the sheet has closed
+ *
+ */
+export class PharosSheet extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-button': PharosButton,
+    'pharos-heading': PharosHeading,
+    'focus-trap': FocusTrap,
+  };
+
+  /**
+   * Indicates if the sheet is open.
+   * @attr open
+   */
+  @property({ type: Boolean, reflect: true })
+  public open = false;
+
+  /**
+   * Indicates if the sheet is expanded.
+   * @attr expanded
+   */
+  @property({ type: Boolean, reflect: true })
+  public expanded = false;
+
+  /**
+   * Indicates if the sheet is allowed to expand.
+   * @attr enbaleExpansion
+   */
+  @property({ type: Boolean, reflect: true })
+  public enbaleExpansion = true;
+
+  /**
+   * Indicates if the sheet contains close button.
+   * @attr hasClose
+   */
+  @property({ type: Boolean, reflect: true })
+  public hasClose = true;
+
+  /**
+   * Text content for the sheet header
+   * @attr header
+   */
+  @property({ type: String, reflect: true })
+  public header = 'Sheet header';
+
+  /**
+   * Indicates if the sheet footer should contain a divider.
+   * @attr footer-divider
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'footer-divider' })
+  public footerDivider = false;
+
+  private _currentTrigger: Element | null = null;
+
+  private _triggers!: NodeListOf<HTMLElement>;
+
+  @state()
+  private _isFooterEmpty = true;
+
+  private _startHeight = 0;
+  private _startY = 0;
+  private _isDragging = false;
+
+  constructor() {
+    super();
+    this._handleKeydown = this._handleKeydown.bind(this);
+    this._handleTriggerClick = this._handleTriggerClick.bind(this);
+    if (this.enbaleExpansion) {
+      this.addEventListener('touchend', this._handleDragEnd);
+      this.addEventListener('mouseup', this._handleDragEnd);
+      this.addEventListener('touchmove', this._handleTouchDragging);
+      this.addEventListener('mousemove', this._handleMouseDragging);
+    }
+  }
+
+  public static override get styles(): CSSResultArray {
+    return [sheetStyles];
+  }
+
+  protected override firstUpdated(): void {
+    this._addTriggerListeners();
+  }
+
+  protected override update(changedProperties: PropertyValues): void {
+    super.update && super.update(changedProperties);
+  }
+
+  protected override updated(changedProperties: PropertyValues): void {
+    if (changedProperties.has('open')) {
+      const body = document.querySelector('body');
+
+      if (this.open) {
+        body?.classList.add('pharos-sheet__body');
+        this._focusContents();
+      } else {
+        body?.classList.remove('pharos-sheet__body');
+        this._returnTriggerFocus();
+      }
+
+      if (changedProperties.get('open') !== undefined) {
```

Comment:
is this condition not covered by the upper closure?
https://github.com/ithaka/pharos/pull/603/files#diff-1f2222a8dc07b2a9918b11299ed4578fa635f936cb9b05ff423597cc0ef8d3e1R117

Reply:
@michael-iden do you know why do we add the additional checking here?

Reply:
We should ensure only code we need is being used, and a fundamental understanding of what's going on with each bit would be a good place to start. Lit's [Reactive properties](https://lit.dev/docs/components/properties/) is a good reference.

Reply:
Ah so from what I see in the doc, "changedProperties is a Map with keys that are the names of changed properties and values that are the corresponding previous values". So `changedProperties.has('open')` just check if the map has key `open` but `changedProperties.get('open')` checks the actual value of `open`?